internal/controllers/aggregation/slice.go

```diff
@@ -36,7 +36,15 @@ func (s *sliceController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	if err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("getting composition: %w", err))
 	}
-	logger = logger.WithValues("compositionGeneration", comp.Generation, "compositionName", comp.Name, "compositionNamespace", comp.Namespace)
+
+	synthesisID := ""
+	if comp.Status.CurrentSynthesis != nil {
```

Comment:
Would you mind moving this logic to a getter-style func on the composition status struct? To avoid duplicating the logic across controllers.

Like: `func (s *CompositionStatus) GetCurrentSynthesisUUID() string {`

Reply:
(just a nit, feel free to merge without addressing)

Reply:
Updated.